src/test/java/com/team701/buddymatcher/services/pairing/PairingServiceImplUnitTests.java

```diff
@@ -0,0 +1,108 @@
+package com.team701.buddymatcher.services.pairing;
+
+import com.team701.buddymatcher.domain.users.Buddies;
+import com.team701.buddymatcher.domain.users.User;
+import com.team701.buddymatcher.repositories.users.UserRepository;
+import com.team701.buddymatcher.repositories.users.BuddiesRepository;
+import com.team701.buddymatcher.services.pairing.impl.PairingServiceImpl;
+import com.team701.buddymatcher.services.users.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.*;
+
+
+@ExtendWith(MockitoExtension.class)
+public class PairingServiceImplUnitTests {
+
+    @Mock
+    private BuddiesRepository buddiesRepository;
+
+    @Mock
+    private UserService userService;
+
+    @InjectMocks
+    private PairingServiceImpl pairingService;
+
+    @Test
+    void correctlyAddBuddy(){
+
+        Long user_1_Id = new Random().nextLong();
+        Long buddies_1_Id = new Random().nextLong();
+        List buddies_1_list = new ArrayList();
+        Buddies buddies_1 = new Buddies().setId(buddies_1_Id).setUsers(buddies_1_list);
+        User user_1 = createExpectedUser_1(user_1_Id, buddies_1);
+
+        Long user_2_Id = new Random().nextLong();
+        Long buddies_2_Id = new Random().nextLong();
+        List buddies_2_list = new ArrayList();
+        Buddies buddies_2 = new Buddies().setId(buddies_2_Id).setUsers(buddies_2_list);
+        User user_2 = createExpectedUser_2(user_2_Id,buddies_2);
+
+        Mockito.when(userService.retrieve(user_1_Id)).thenReturn(user_1);
+        Mockito.when(userService.retrieve(user_2_Id)).thenReturn(user_2);
```

Comment:
similar to the previous comment,  `retrieve` will need to be changed to `retrieveById`